### PR TITLE
Power loss conditions

### LINKED ISSUE #495: Check that subsystems respond gracefully to power loss conditions
Power cycles and power dropouts necessitate more careful tracking of these bugs.
- Power cycling creates a delay between an output going off and coming back online

For example:
- QuakeManager needs to check the "powered" flag prior to doing comms
- Do ADCS, prop, and Piksi have this problem? #487 tracks an ADCS issue where it takes some time to ramp up. But the other issues don't have associated tickets.

### PR DESCRIPTION
**Pull request by afoarce**
# Power Loss Conditions

Fixes #495 

### Summary of changes
- Update Quake to check for power cycling command before leaving wait and update QuakeFaultHandler to ensure power cycles only start when in wait 
- Update Prop to check for is_functional() before state changes
- Add loop() to ADCS to populate state fields before I2C begins 
- Add unit tests for the above changes

### Testing

Added unit tests for the changes added and ensured old unit tests pass correctly with the modifications.

### Documentation Evidence
Overall small modifications that are documented in line and in the unit tests.

### CODE REVIEW

File: src/fsw/FCCode/Drivers/PropulsionSystem.cpp [hunk 1]
```diff
@@ -57,7 +57,11 @@ int _Tank2::get_fake_temp_analog() const
 
 /** Initialize static variables */
 
-_PropulsionSystem::_PropulsionSystem() : Device("propulsion") {}
+_PropulsionSystem::_PropulsionSystem() : Device("propulsion") {
+    #ifdef UNIT_TEST
+    test_val = true;
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/Drivers/PropulsionSystem.cpp:62]: Might be worth making the name of this variable more descriptive. Maybe something like "fake_is_functional"

Also @athena255 had a nice naming convention for these things that I'm forgetting

**Review by tanishqaggarwal — APPROVED** [review]
Looks good to me. Just one minor comment

**Comment by tanishqaggarwal** [comment]
I'd recommend grabbing @athena255 's review too since she wrote both quake and prop

**Review by athena255 — APPROVED** [review]
This looks good.

**Pull request merged**